src/encoded/static/components/auditmatrix.js

```diff
@@ -0,0 +1,435 @@
+        this.setState((prevState) => {
+            const newState = {};
+
+            // If the See All button wasn't open (meaning this function was called because the user
+            // wanted to see all), forget all the individual ones the user had opened so that
+            // they're all closed when the user clicks See Fewer.
+            if (!prevState.allYGroupsOpen) {
+                const groupOpen = {};
+                Object.keys(prevState.yGroupOpen).forEach((key) => {
+                    groupOpen[key] = false;
+                });
+                newState.yGroupOpen = groupOpen;
+            }
+
+            // Toggle the state of allYGroupsOpen.
+            newState.allYGroupsOpen = !prevState.allYGroupsOpen;
+            return newState;
+        });
+    }
+
+    render() {
+        const context = this.props.context;
+        const matrix = context.matrix;
+        const parsedUrl = url.parse(this.context.location_href);
+        const matrixBase = parsedUrl.search || '';
+        const matrixSearch = matrixBase + (matrixBase ? '&' : '?');
+        const notification = context.notification;
+        const visualizeLimit = 500;
+        if (notification === 'Success' || notification === 'No results found') {
+            const xFacets = matrix.x.facets.map(f => _.findWhere(context.facets, { field: f })).filter(f => f);
+            let yFacets = matrix.y.facets.map(f => _.findWhere(context.facets, { field: f })).filter(f => f);
+            yFacets = yFacets.concat(_.reject(context.facets, f => _.contains(matrix.x.facets, f.field) || _.contains(matrix.y.facets, f.field)));
+            const xGrouping = matrix.x.group_by;
+            const primaryYGrouping = matrix.y.group_by[0];
+            const secondaryYGrouping = matrix.y.group_by[1];
+            const xBuckets = matrix.x.buckets;
+            const xLimit = matrix.x.limit || xBuckets.length;
+            let yGroups = matrix.y[primaryYGrouping].buckets;
+            // The following lines are to make sure that the audit categories are in the correct
+            // order and to assign a proper title to each colored row.
+            const orderKey = ['no_audits', 'audit.WARNING.category', 'audit.NOT_COMPLIANT.category',
+                'audit.ERROR.category', 'audit.INTERNAL_ACTION.category'];
+            const titleKey = ['No audits', 'Warning', 'Not Compliant', 'Error', 'Internal Action'];
+            const noAuditKey = ['no errors, compliant, and no warnings', 'no errors and compliant',
+                'no errors', 'no audits'];
+            // For each group, compare against the key arrays above and format yGroups so that
+            // it has the same order as the keys and each group in yGroups has the correct title.
+            let orderIndex = 0;
+            let rowIndex = 0;
+            const tempYGroups = [];
+            const tempNoAudits = [];
+            while (orderIndex < orderKey.length) {
+                yGroups.forEach((group) => {
```

Comment:
There seems to be [debate](https://github.com/eslint/eslint/issues/5044) about this rule, and they did put in a change to reduce the number of false positives, but all the way back in January 2016.

I still think this warning is wrong because, for the inner loop with the same warning, you can get rid of the warning simply by not using `rowIndex` inside the loop — but that reference is needed. I’m betting you can get around this error by writing this quadruple-nested loop in a different way, but that would take some thinking through.

I recommend having @kjain21 just disable this warning. She told me she’ll see if she can come up with another way to do this code, and disable the warning if not.